cmd/edv-rest/startcmd/start.go

```diff
@@ -162,26 +161,24 @@ func startEDV(parameters *edvParameters) error {
 	return err
 }
 
-func createProvider(parameters *edvParameters) (storage.Provider, error) {
-	var provider storage.Provider
+func createEDVProvider(parameters *edvParameters) (edvprovider.EDVProvider, error) {
+	var edvProv edvprovider.EDVProvider
 
 	switch {
 	case strings.EqualFold(parameters.databaseType, databaseTypeMemOption):
-		provider = memstore.NewProvider()
+		edvProv = memedvprovider.NewProvider()
+
+		log.Warn("encrypted indexing and querying is disabled since they are not supported by memstore")
 	case strings.EqualFold(parameters.databaseType, databaseTypeCouchDBOption):
-		couchDBProvider, err := couchdbstore.NewProvider(parameters.databaseURL)
+		couchDBEDVProv, err := couchdbedvprovider.NewProvider(parameters.databaseURL)
 		if err != nil {
-			if err.Error() == "hostURL for new CouchDB provider can't be blank" {
-				return nil, errMissingDatabaseURL
-			}
-
-			return nil, err
+			return edvProv, err
```

Comment:
Any reason why we are returning provider and error?

Reply:
It should still be nil either way, but yeah it's probably better to just return nil directly. Done.